ha_mroonga.cpp

```diff
@@ -12374,10 +12374,11 @@ int ha_mroonga::generic_store_bulk_year(Field* field, grn_obj* buf)
   int error = 0;
   bool truncated = false;
 
-  int year = 1970;
+  int year = mrn::TimeConverter::EPOCH_YEAR;
+  mrn::TimeConverter time_converter;
   if (!field->is_null()) {
     if (field->field_length == 2) {
-      year = static_cast<int>(field->val_int() + 2000);
+      year = time_converter.two_digits_year_to_mysql_year(field->val_int());
```

Comment:
由来のコメントを書いておけば十分じゃないかな。

```suggestion
      const auto two_digits_year = field->val_int();
      // https://dev.mysql.com/doc/refman/9.0/en/year.html
      // As 1- or 2-digit numbers in the range 0 to 99. MySQL converts
      // values in the ranges 1 to 69 and 70 to 99 to YEAR values in the
      // ranges 2001 to 2069 and 1970 to 1999. 
      if (two_digits_year < 70) {
        year = 2000 + two_digits_year;
      else {
        year = 1900 + two_digits_year;
      }
```